### PR TITLE
Add new `Minitest/UnspecifiedException` cop

### PR DESCRIPTION
**Pull request by fatkodima**
To avoid false-positives when checking exceptions, it is valuable to be able to enforce that each `assert_raises` includes specific exception(s) it checks against.

### CODE REVIEW

File: docs/modules/ROOT/pages/cops_minitest.adoc [hunk 1]
```diff
@@ -785,3 +785,30 @@ refute_respond_to(self, :do_something)
...
+
+This cop checks for a specified error in `assert_raises`.
+
+=== Examples
+
+[source,ruby]
+----
+# bad
+assert_raises { do_something}
+assert_raises('message') { do_something}
```
Comment (koic) [inline docs/modules/ROOT/pages/cops_minitest.adoc:809]: I think this case can be allowed like RuboCop RSpec.
https://github.com/rubocop-hq/rubocop-rspec/blob/v1.41.0/lib/rubocop/cop/rspec/unspecified_exception.rb#L24-L26

**Comment by koic** [comment]
This proposal looks good to me. Can you open a PR to the style guide for this rule?

File: docs/modules/ROOT/pages/cops_minitest.adoc (hunk 1 above)
Comment (fatkodima) [reply docs/modules/ROOT/pages/cops_minitest.adoc:809]: Minitest does not accept strings as exception messages, only exception classes - http://docs.seattlerb.org/minitest/Minitest/Assertions.html#method-i-assert_raises

It only accepts string message as last argument.

**Comment by fatkodima** [comment]
Opened https://github.com/rubocop-hq/minitest-style-guide/pull/33 and added `StyleGuide` section to that.
Changed cop class example with `'message'`, to reflect that it is not related to `Exception#message`, as opposite to rspec.

File: docs/modules/ROOT/pages/cops_minitest.adoc [hunk 2]
```diff
@@ -785,3 +785,34 @@ refute_respond_to(self, :do_something)
...
+|===
+
+This cop checks for a specified error in `assert_raises`.
+
+=== Examples
+
+[source,ruby]
+----
+# bad
+assert_raises { do_something}
```
Comment (abhaynikam) [inline docs/modules/ROOT/pages/cops_minitest.adoc:808]: Space after `do_something` was missing. Maybe we could rename `do_something` to `do_something_that_raises_exception`. Just more readable 😊 

```suggestion
assert_raises { do_something }
```

Comment (fatkodima) [reply docs/modules/ROOT/pages/cops_minitest.adoc]: Thanks! Changed it to `raise FooException`.

**Comment by koic** [comment]
Thanks!

**Pull request merged**